### DISCUSSION TITLE
Conversion from .h5 to .csv formats

### DISCUSSION THREAD

**Discussion by sahilsingh2402** [talmolab/sleap#1333] [Help!] [answered]
Hi, I am very new to SLEAP, working on making behavioral classifiers on flies. How can I convert from .h5 file format to .csv file format? Also, I am using a very limited GPU device, so it takes a lot of time for me to train even on single batch size, any tips to speedup?

**Comment by roomrys** [comment, marked as answer]
Hi @sahilsingh2402,

Another SLEAP user (and contributor!) @Toshea111 has created [this notebook](https://github.com/Toshea111/sleap/blob/develop/docs/notebooks/Convert_HDF5_to_CSV_updated.ipynb) for converting SLEAP files to a CSV format.

I am working on integrating this adaptor into SLEAP for a future release (estimated 1.3.2 - currently 1.3.0), but until then the notebook is a wonderful outside contribution to SLEAP that should do the job!

Thanks,
Liezl

---
Related #1089

> I have provided an updated version of the notebook here:
> 
> https://github.com/Toshea111/sleap/blob/develop/docs/notebooks/Convert_HDF5_to_CSV_updated.ipynb
> 
> It fixes some issues with compatibility and illegal slicing arguments in the original version.
> 
> Best wishes,
> 
> Thomas

**Reply by sahilsingh2402** [reply]
Thanks a lot for this :)
